### PR TITLE
feat(docs): Added reference section

### PR DESCRIPTION
**Pull request by fhennig**
# Description

see title

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/zookeeper/pages/configuration.adoc [hunk 1]
```diff
@@ -1,17 +1,60 @@
...
+  name: simple
+spec:
+  image:
+    productVersion: 3.8.3
+  servers:
+    roleGroups:
+      default:
+        selector:
+          matchLabels:
+            kubernetes.io/os: linux
```
Comment (sbernauer) [inline docs/modules/zookeeper/pages/configuration.adoc:21]:
```suggestion
```

**Review by sbernauer — CHANGES_REQUESTED** [review]
LGTM otherwise

File: docs/modules/zookeeper/pages/configuration.adoc [hunk 2]
```diff
@@ -1,17 +1,60 @@
...
-== Environment variables
-
-This operator accepts the following environment variables:
-
-include::env_var_args.adoc[]
-
 == Kubernetes custom resource options
 
-include::config_properties.adoc[]
+The cluster can be configured via a YAML file. This custom resource specifies the amount of replicas for each role group or role specific configuration like port definitions etc.
```
Comment (maltesander) [inline docs/modules/zookeeper/pages/configuration.adoc:5]: I dont see port definitions in the example. Maybe reference the tickTime?

File: docs/modules/zookeeper/pages/usage_guide/authentication.adoc [hunk 3]
```diff
@@ -1,8 +1,7 @@
 = Authentication
 
-The communication between nodes (server to server) is encrypted TLS by default. In order to enforce TLS authentication
-for client-to-server communication, you can set an `AuthenticationClass` reference in the custom resource provided by
-the xref:commons-operator:index.adoc[Commons Operator].
+The communication between nodes (server to server) is encrypted TLS by default.
```
Comment (maltesander) [inline docs/modules/zookeeper/pages/usage_guide/authentication.adoc:3]:
```suggestion
The communication between nodes (server to server) is encrypted via TLS by default.
```

**Review by maltesander — CHANGES_REQUESTED** [review]

File: docs/modules/zookeeper/pages/configuration.adoc (hunk 2 above)
Comment (fhennig) [reply docs/modules/zookeeper/pages/configuration.adoc]: I forgot to delete the file (it's now deleted). All the info is in the CRD docs

**Review by maltesander — APPROVED** [review]
LGTM!

**Review by sbernauer — APPROVED** [review]
Thanks!

**Pull request merged**